### PR TITLE
[webgl] Modularize batchMatMul, sum.

### PR DESCRIPTION
**Pull request by annxingyuan**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4140)

### CODE REVIEW

**Review by jinjingforever — APPROVED** [review]
LGTM. Thanks Ann!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4140#-:-MKp5SS5Ewfv5q8SzsYW:bovic7g)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @annxingyuan, @jinjingforever, @lina128, and @tafsiri)

___
*[tfjs-backend-webgl/src/register_all_kernels.ts, line 68 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4140#-MKp3SpnAdP9WZVSMGzU:-MKp3Spo0A1NLg0BSoK4:b-bhyaqf) ([raw file](https://github.com/tensorflow/tfjs/blob/7a441c33d8b73eaf928b7417d951f6111e062911/tfjs-backend-webgl/src/register_all_kernels.ts#L68)):*
> ```XML
>   avgPoolBackpropConfig,
>   batchNormConfig,
>   batchMatMulConfig,
> ```

nit: move above batchNormConfig to keep the order

**Review by tafsiri — CHANGES_REQUESTED** [review]
Thanks Ann! 

> Reviewed 9 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4140#-:-MKpPb4n1e7ajjBSlCvb:b727nzi)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @annxingyuan, @lina128, and @tafsiri)

___
*[tfjs-backend-webgl/src/kernels/_FusedMatMul.ts, line 45 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4140#-MKpNHgPAlC8QheYggCe:-MKpNHgQ9nsjpsOUXwUI:b-lk7dxs) ([raw file](https://github.com/tensorflow/tfjs/blob/7a441c33d8b73eaf928b7417d951f6111e062911/tfjs-backend-webgl/src/kernels/_FusedMatMul.ts#L45)):*
> ```XML
> 
> export const _fusedMatMulConfig: KernelConfig = {
>   kernelName: _FusedMatMul,
> ```

[non blocking] did we decide to remove the underscore prefixes? (honestly can't remember. if so this might be a good PR to do that across the board).
___
*[tfjs-backend-webgl/src/kernels/Sum_impl.ts, line 24 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4140#-MKpOggQDYUTJRGfdIPi:-MKpOggQDYUTJRGfdIPj:bu7pho9) ([raw file](https://github.com/tensorflow/tfjs/blob/7a441c33d8b73eaf928b7417d951f6111e062911/tfjs-backend-webgl/src/kernels/Sum_impl.ts#L24)):*
> ```XML
> import {reshape} from '../kernels/Reshape';
> 
> export function sumImpl(
> ```

Unlike the other impl functions this one can't be directly subbed in as a replacement for sum in other kernels (it doesn't handle the axis and keepDims). How would you see this being reused?

**Review by annxingyuan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4140)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @lina128 and @tafsiri)

___
*[tfjs-backend-webgl/src/register_all_kernels.ts, line 68 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4140#-MKp3SpnAdP9WZVSMGzU:-MKpSET7-6YU8r0_7d3U:bb6un1j) ([raw file](https://github.com/tensorflow/tfjs/blob/7a441c33d8b73eaf928b7417d951f6111e062911/tfjs-backend-webgl/src/register_all_kernels.ts#L68)):*
<details><summary><i>Previously, jinjingforever (Jing Jin) wrote…</i></summary><blockquote>

nit: move above batchNormConfig to keep the order
</blockquote></details>

Done
___
*[tfjs-backend-webgl/src/kernels/_FusedMatMul.ts, line 45 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4140#-MKpNHgPAlC8QheYggCe:-MKpSFOw-HW2FbYrO75m:b-te2oic) ([raw file](https://github.com/tensorflow/tfjs/blob/7a441c33d8b73eaf928b7417d951f6111e062911/tfjs-backend-webgl/src/kernels/_FusedMatMul.ts#L45)):*
<details><summary><i>Previously, tafsiri (Yannick Assogba) wrote…</i></summary><blockquote>

[non blocking] did we decide to remove the underscore prefixes? (honestly can't remember. if so this might be a good PR to do that across the board).
</blockquote></details>

I'd say let's do that in a separate PR now that we know it would entail changing the converter as well (since the underscore was introduced to accommodate grappler behavior)
___
*[tfjs-backend-webgl/src/kernels/Sum_impl.ts, line 24 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4140#-MKpOggQDYUTJRGfdIPi:-MKpTvWi0mN1PYjBe0si:bb6un1j) ([raw file](https://github.com/tensorflow/tfjs/blob/7a441c33d8b73eaf928b7417d951f6111e062911/tfjs-backend-webgl/src/kernels/Sum_impl.ts#L24)):*
<details><summary><i>Previously, tafsiri (Yannick Assogba) wrote…</i></summary><blockquote>

Unlike the other impl functions this one can't be directly subbed in as a replacement for sum in other kernels (it doesn't handle the axis and keepDims). How would you see this being reused?
</blockquote></details>

Done

**Review by lina128 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4140)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @lina128 and @tafsiri)

___
*[tfjs-backend-webgl/src/kernels/_FusedMatMul.ts, line 45 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4140#-MKpNHgPAlC8QheYggCe:-MKpuTA--Dj3m4EJp6QX:b5vvr9j) ([raw file](https://github.com/tensorflow/tfjs/blob/7a441c33d8b73eaf928b7417d951f6111e062911/tfjs-backend-webgl/src/kernels/_FusedMatMul.ts#L45)):*
<details><summary><i>Previously, annxingyuan (Ann Yuan) wrote…</i></summary><blockquote>

I'd say let's do that in a separate PR now that we know it would entail changing the converter as well (since the underscore was introduced to accommodate grappler behavior)
</blockquote></details>

I agree with doing it in a separate PR, but I think we don't need to change converter. The op name in converter doesn't have to be consistent with kernel name in core, they don't align in today's code anyway, for example FusedConv2D (it's _FusedConv2D in converter). As long as we still do mapping (in executors and kernel2op.json), we can decouple the naming, right?

**Review by tafsiri — APPROVED** [review]
> Reviewed 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4140)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @annxingyuan and @lina128)

___
*[tfjs-backend-webgl/src/kernels/_FusedMatMul.ts, line 45 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4140#-MKpNHgPAlC8QheYggCe:-MKpvl_R1aYoQO0veA_O:b-6ul73b) ([raw file](https://github.com/tensorflow/tfjs/blob/7a441c33d8b73eaf928b7417d951f6111e062911/tfjs-backend-webgl/src/kernels/_FusedMatMul.ts#L45)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

I agree with doing it in a separate PR, but I think we don't need to change converter. The op name in converter doesn't have to be consistent with kernel name in core, they don't align in today's code anyway, for example FusedConv2D (it's _FusedConv2D in converter). As long as we still do mapping (in executors and kernel2op.json), we can decouple the naming, right?
</blockquote></details>

+1 on both counts. There shouldn't currently be a dependency between the string in converter and the kernel called in core (it should be going through the op). If there is a dependency we should make sure it uses the constant and not the raw string.
